libs/community/langchain_community/document_loaders/parsers/pdf.py

```diff
@@ -1160,50 +1160,221 @@ def _extract_tables_from_page(self, page: pymupdf.Page) -> str:
 
 
 class PyPDFium2Parser(BaseBlobParser):
-    """Parse `PDF` with `PyPDFium2`."""
+    """Parse a blob from a PDF using `PyPDFium2` library.
 
-    def __init__(self, extract_images: bool = False) -> None:
-        """Initialize the parser."""
+    This class provides methods to parse a blob from a PDF document, supporting various
+    configurations such as handling password-protected PDFs, extracting images, and
+    defining extraction mode.
+    It integrates the 'PyPDFium2' library for PDF processing and offers synchronous
+    blob parsing.
+
+    Examples:
+        Setup:
+
+        .. code-block:: bash
+
+            pip install -U langchain-community pypdfium2
+
+        Load a blob from a PDF file:
+
+        .. code-block:: python
+
+            from langchain_core.documents.base import Blob
+
+            blob = Blob.from_path("./example_data/layout-parser-paper.pdf")
+
+        Instantiate the parser:
+
+        .. code-block:: python
+
+            from langchain_community.document_loaders.parsers import PyPDFium2Parser
+
+            parser = PyPDFium2Parser(
+                # password=None,
+                mode="page",
+                pages_delimiter="\n\f",
+                # extract_images = True,
+                # images_to_text = convert_images_to_text_with_tesseract(),
+            )
+
+        Lazily parse the blob:
+
+        .. code-block:: python
+
+            docs = []
+            docs_lazy = parser.lazy_parse(blob)
+
+            for doc in docs_lazy:
+                docs.append(doc)
+            print(docs[0].page_content[:100])
+            print(docs[0].metadata)
+    """
+
+    # PyPDFium2 is not thread safe.
+    # See https://pypdfium2.readthedocs.io/en/stable/python_api.html#thread-incompatibility
+    _lock = threading.Lock()
+    warnings.filterwarnings(
```

Comment:
could you do this with a context manager? Library code shouldn't be globally configuring warning filters on a 3rd party

Reply:
It's not possible to reset the filter warning.
```
    class _WarningFilter(object):
        def __enter__(self):
            warnings.filterwarnings(
                "ignore",
                module=r"^pypdfium2._helpers.textpage$",
                message="get_text_range\\(\\) call with default params will be .*",
            )

        def __exit__(self, type, value, traceback):
            warnings.resetwarnings()  # Reset ALL filter
```
I prefer to delete this